src/gleam/string.gleam

```diff
@@ -95,40 +95,87 @@ pub external fn uppercase(String) -> String = "string" "uppercase"
 pub external fn compare(String, String) -> order.Order =
   "gleam_stdlib" "compare_strings"
 
-/// ## Get Substrings
+// slice helper
+external fn erl_string_slice(string, start, length) -> String =
+  "string" "slice"
 
-// TODO
-/// Take a substring given a start and end Grapheme indexes. Negative indexes
-/// are taken starting from the *end* of the list.
+/// ## Get Substrings
+///
+/// Take a substring given a starting index and length.
+/// A negative start or length will return an empty string.
+/// Works on Unicode Graphemes.
 ///
 /// ## Examples
 /// ```gleam
-/// slice("gleam", from: 1, to: 3) == "lea"
-/// slice("gleam", from: 1, to: 10) == "leam"
-/// slice("snakes on a plane!", from: -6, to: -1) == "plane"
+/// slice("gleam", from: 1, length: 3) == "lea"
+/// slice("gleam", from: 1, length: 10) == "leam"
```

Comment:
The label name length doesn't fit with the naming convention used so far. "up_to" perhaps? What do you think?

Reply:
it is a length. Would you rather slice took two indexes rather than a start index and a length?